.github/workflows/docker.yml

```diff
@@ -0,0 +1,51 @@
+name: Docker
+
+on:
+  push:
+    tags:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Clone Repository
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.BOT_PAT }}
+          submodules: recursive
```

Comment:
I think these are unnecessary; the workflow worked [in my fork](https://github.com/jamesbornholt/rosette/runs/5402347629?check_suite_focus=true) without a PAT.